src/coordinate_search.jl

```diff
@@ -0,0 +1,77 @@
+export coordinate_search
+"""
+	coordinate_search(nlp)
+	The coordinate search is a derivative free optimization method.
+	It minimizes a certain function f by walking on a grid and taking a step towards smaller function values.
+	When it can no longer find any smaller values, it reduces the grid by a factor of β, repeating the process.
+"""
+function coordinate_search(nlp :: AbstractNLPModel;
+			                     x :: AbstractVector=copy(nlp.meta.x0),
+                           tol :: Real = min(√eps(eltype(x)), 1e-4),
+                           α :: Real = 1.0,
+                           β :: Real = 2.0,
+			                     max_time :: Float64  = 30.0,
+			                     max_eval :: Int = -1,
+			                     greedy :: Bool = true)
+
+  k = 0
+  el_time = 0.0
+  start_time = time()
+  tired = neval_obj(nlp) > max_eval >= 0 || el_time > max_time
+  optimal = α < tol
+  T = eltype(x)
+  status =:unknown
+  @info log_header([:iter, :f, :H],[Int, T, T], hdr_override=Dict(:f=>"f(x)", :H=>"α"))
+
+  f = obj(nlp,x)
+  best_f = f
+  best_i = 0
+  while !(optimal || tired)
+    @info log_row(Any[k, f, α])
+    xt = copy(x)
+    success = done = false
+    while !(done)
+      for i in 1:nlp.meta.nvar,  s in [-1,1]
+	      xt[i] = x[i] + α * s
+	      f_xt = obj(nlp,xt)
+	      if (f_xt < best_f)
+					success = true
+	      	best_f = f_xt
+	  			best_i = i * s
+	  			if greedy
+	    			done = true
+	    			break
+	  		end
+			end
+        xt[i] = x[i]
+    	end
+    	done = true
+    end

```

Comment:
Fix indentation

Reply:
You don't need `done` nor the `while`. Line 48 always happens because it's outside the for loop, so just remove the `while`.